### PR TITLE
test: Have added negative test cases for new folders

### PR DESCRIPTION
**Pull request by ShyamOOAI**
## Summary by CodeRabbit

- **Tests**
	- Added comprehensive test cases for dashboard and alerts folder functionalities.
	- Introduced tests for unauthorized access scenarios.
	- Added error handling tests for invalid inputs and folder IDs.
	- Improved test coverage for various edge cases, including responses for empty and invalid folder IDs.

### CODE REVIEW

File: tests/api-testing/tests/test_foldersnew.py [hunk 1]
```diff
@@ -128,12 +199,62 @@ def test_new_create_alerts_update_delete_folders(create_session, base_url):
...
+    resp_create_new_alerts_folders_blank = create_session.post(
+        f"{base_url}api/v2/{org_id}/folders/alerts", json=payload
+    )
+    print(create_session.headers)
+    assert resp_create_new_alerts_folders_blank.status_code == 400, (
+        f"Expected 400, but got {resp_create_new_alerts_folders_blank.status_code} "
+        f"{resp_create_new_alerts_folders_blank.content}"
+    )
+
+def test_retrieve_alerts_folders_400(create_session, base_url):
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/api-testing/tests/test_foldersnew.py:228]: have updated

File: tests/api-testing/tests/test_foldersnew.py [hunk 2]
```diff
@@ -128,12 +199,62 @@
...
+    invalid_folder_id = "e56454545dfd@43232"  # Folder ID that doesn't exist
+    
+    resp_get_alerts_folders_invalid = create_session.get(
+        f"{base_url}api/v2/{org_id}/folders/dashboards/{invalid_folder_id}"
+    )
+    assert resp_get_alerts_folders_invalid.status_code == 404, (
+        f"Expected 404 Not Found, but got {resp_get_alerts_folders.status_code} "
+        f"{resp_get_alerts_folders_invalid.content}"
+    )
+
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/api-testing/tests/test_foldersnew.py:253]: have updated

File: tests/api-testing/tests/test_foldersnew.py [hunk 3]
```diff
@@ -128,12 +199,62 @@ def test_new_create_alerts_update_delete_folders(create_session, base_url):
...
+        f"Expected 404 , but got {resp_create_new_alerts_folders_blank.status_code} "
+        f"{resp_create_new_alerts_folders_blank.content}"
+    )
+
+def test_invalid_folder_id_retrieve_alerts(create_session, base_url):
+    org_id = "default"
+    invalid_folder_id = "e56454545dfd@43232"  # Folder ID that doesn't exist
+    
+    resp_get_alerts_folders_invalid = create_session.get(
+        f"{base_url}api/v2/{org_id}/folders/dashboards/{invalid_folder_id}"
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/api-testing/tests/test_foldersnew.py:247]: have updated

File: tests/api-testing/tests/test_foldersnew.py [hunk 4]
```diff
@@ -128,12 +199,62 @@
...
+    )
+    assert resp_get_alerts_folders_invalid.status_code == 404, (
+        f"Expected 404 Not Found, but got {resp_get_alerts_folders.status_code} "
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/api-testing/tests/test_foldersnew.py:250]: have updated

File: tests/api-testing/tests/test_foldersnew.py [hunk 5]
```diff
@@ -128,12 +199,62 @@
...
+    # Use create_session instead of session
+    resp_create_new_alerts_folders_blank = create_session.post(
+        f"{base_url}api/v2/{org_id}/folders/alerts", json=payload
+    )
+    print(create_session.headers)
+    assert resp_create_new_alerts_folders_blank.status_code == 404, (
+        f"Expected 404 , but got {resp_create_new_alerts_folders_blank.status_code} "
+        f"{resp_create_new_alerts_folders_blank.content}"
+    )
+
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/api-testing/tests/test_foldersnew.py:241]: have updated

File: tests/api-testing/tests/test_foldersnew.py [hunk 6]
```diff
@@ -128,12 +199,62 @@ def test_new_create_alerts_update_delete_folders(create_session, base_url):
...
+def test_retrieve_alerts_folders_404_empty_org(create_session, base_url):
+    org_id = "" # Empty org_id to test the case
+    payload = {"description": "newfoldernvp", "name": ""}
+    
+    # Use create_session instead of session
+    resp_create_new_alerts_folders_blank = create_session.post(
+        f"{base_url}api/v2/{org_id}/folders/alerts", json=payload
+    )
```
Comment (patrickcsullivan) [inline tests/api-testing/tests/test_foldersnew.py:235]: I'm not sure if this is the case or not, but I would expect that calling the `CreateFolder` route with `"name"` set to the empty string should result in a 400 Bad Request error.

File: tests/api-testing/tests/test_foldersnew.py [hunk 7]
```diff
@@ -128,12 +199,62 @@ def test_new_create_alerts_update_delete_folders(create_session, base_url):
...
+    print(create_session.headers)
+    assert resp_create_new_alerts_folders_blank.status_code == 400, (
+        f"Expected 400, but got {resp_create_new_alerts_folders_blank.status_code} "
+        f"{resp_create_new_alerts_folders_blank.content}"
+    )
```
Comment (patrickcsullivan) [inline tests/api-testing/tests/test_foldersnew.py:226]: nit: The name of this test is `_retrieve_alerts_folders_` but it seems like it's just testing the creation of alert folders rather than the retrieval of alert folders.

File: tests/api-testing/tests/test_foldersnew.py [hunk 8]
```diff
@@ -66,6 +63,80 @@ def test_new_create_update_delete_dashboards_folders(create_session, base_url):
...
+    
+    # Use create_session instead of session
+    resp_create_new_dashboards_folders_blank = create_session.post(
+        f"{base_url}api/v2/{org_id}/folders/dashboards", json=payload
+    )
+    print(create_session.headers)
+    assert resp_create_new_dashboards_folders_blank.status_code == 404, (
+        f"Expected 404 , but got {resp_create_new_dashboards_folders_blank.status_code} "
+        f"{resp_create_new_dashboards_folders_blank.content}"
+    )
```
Comment (patrickcsullivan) [inline tests/api-testing/tests/test_foldersnew.py:116]: nit: The name of this test is `_retrieve_dashboards_folders_` but it seems like it's just testing the creation of dashboard folders rather than the retrieval of dashboard folders.

File: tests/api-testing/tests/test_foldersnew.py [hunk 9]
```diff
@@ -66,6 +63,80 @@ def test_new_create_update_delete_dashboards_folders(create_session, base_url):
...
+    assert resp_get_dashboard_folders_Unauthorize.status_code == 401, (
+        f"Expected 401 Unauthorized Not Found, but got {resp_get_dashboard_folders_Unauthorize.status_code} "
+        f"{resp_get_dashboard_folders_Unauthorize.content}"
+    )
+    
+
+def test_retrieve_dashboard_folders_400(create_session, base_url):
+    org_id = "default"
+    folder_id = ""  # Empty folder_id to test the case
```
Comment (patrickcsullivan) [inline tests/api-testing/tests/test_foldersnew.py:91]: I don't think `folder_id` is actually used in this test.

File: tests/api-testing/tests/test_foldersnew.py (hunk 6 above)
Comment (ShyamOOAI) [reply tests/api-testing/tests/test_foldersnew.py:235]: have added as per suggestion

File: tests/api-testing/tests/test_foldersnew.py (hunk 8 above)
Comment (ShyamOOAI) [reply tests/api-testing/tests/test_foldersnew.py:116]: have updated

File: tests/api-testing/tests/test_foldersnew.py (hunk 9 above)
Comment (ShyamOOAI) [reply tests/api-testing/tests/test_foldersnew.py:91]: have removed folder_id

**Review by patrickcsullivan — APPROVED** [review]
